client/src/js/services/journalGridService.js

```diff
@@ -0,0 +1,338 @@
+      totalCredits = precision.round(totalCredits);
+
+      //TO DO : some error message must be translate
+
+      if (singleEntryError) { journalError.throw('ERR_TXN_SINGLE_ENTRY'); }
+      if (!validationService.isTotalsValid(totalDebits, totalCredits)) { journalError.throw('ERR_TXN_IMBALANCE'); }
+      if (accountError) { broadcastError('Records contain invalid or nonexistant accounts.'); }
+      if (dateError) { broadcastError('Transaction contains invalid dates.'); }
+      if (multipleDatesError) { broadcastError('Transaction trans_date field has multiple dates.'); }
+      if (periodError) { broadcastError('Transaction date does not fall in any valid periods.'); }
+      if (fiscalError) { broadcastError('Transaction date does not fall in any valid fiscal years.'); }
+
+      var hasErrors = (
+          dateError || accountError ||
+          balanceError || singleEntryError ||
+          multipleDatesError || fiscalError ||
+          periodError || !validationService.isTotalsValid(totalDebits, totalCredits)
+      );
+
+      return hasErrors;
+    }
+
+    function packager(record) {
+      var data = {}, cpProperties, prop;
+
+      cpProperties = [
+        'uuid', 'project_id', 'trans_id', 'trans_date', 'period_id', 'description', 'account_id',
+        'credit', 'debit', 'debit_equiv', 'credit_equiv', 'fiscal_year_id', 'currency_id',
+        'deb_cred_id', 'deb_cred_type', 'inv_po_id', 'user_id', 'origin_id', 'cc_id', 'pc_id'
+      ];
+
+      for (prop in record) {
+        if (cpProperties.indexOf(prop) > -1) {
+          if (angular.isDefined(record[prop]) && !isNull(record[prop])) {
+            data[prop] = record[prop];
+          }
+        }
+      }
+
+      // FIXME : This will no longer work if we have non-unique account
+      // numbers.
+      if (record.account_number) { data.account_id = gridService.dataLoaderService.account.get(record.account_number).id; }
+
+      // Transfer values from cc over to posting journal cc_id and pc_id fields
+      // This is because we are doing a join, similar to the account_number field
+      // above.
+      // We check for NaNs because we don't have unique identifers like the account number
+      // for an account.
+      if (record.cc && !Number.isNaN(Number(record.cc))) { data.cc_id = record.cc; }
+      if (record.pc && !Number.isNaN(Number(record.pc))) { data.pc_id = record.pc; }
+
+      // FIXME : Hack to get deletion to work with parser.js
+      // This is probably pretty damn insecure
+      if (record.cc === 'null') { data.cc_id = null; }
+      if (record.pc === 'null') { data.pc_id = null; }
+
+      // FIXME : Review this decision
+      data.project_id = sessionService.project.id;
+      data.origin_id = 4;
+      return data;
+    }
+
+    function sort (a,b) {
+      var x = a[this.sortColumn];
+      var y = b[this.sortColumn];
+      return x > y ? 1 : -1;
+    }
+
+    function doParsing (o) { return JSON.parse(JSON.stringify(o)); }
```

Comment:
This should probably be a more meaning name.  This is a hack to deep-copy a JSON object, without copying any of its references.

I'd call it `clone()` or `copy` or `deepClone()` or something along those lines.